enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,114 @@
+Setting `replace-address-first-boot` results in a desired output, i.e. a node which lost its storage will be replaced during bootstrap.
+Otherwise, setting the flag is safe as Scylla ignores it after bootstrap.
+
+Introducing this enhancement additionally allows for removing replacement functionality from `OrphanedPVController`. The controller would be simplified and only be responsible for removing the orphaned PVs.
+
+### User Stories
+
+#### Story 1
+
+The user is running a ScyllaCluster and removes a PVC bound to one of Scylla nodes. The node gets automatically replaced. 
+
+#### Story 2
+
+The user is running a ScyllaCluster on AWS/GCP cluster with a local storage. One of the instances gets restarted, which results in erasing the local storage.
+Therefore, a node scheduled on this instance loses its storage, even though a PVC bound to it is not removed. The node gets replaced automatically.
+
+### Notes/Constraints/Caveats [Optional]
+
+N/A
+
+### Risks and Mitigations
+
+- The proposed design is based on a fact that `replace-address-first-boot` flag is ignored after bootstrap. Any changes to this behaviour or side effects introduced could result in an undefined behaviour of the enhancement.
+
+
+## Design Details
+
+- The InitContainer uses the scylla-operator image.
+- In order to trigger running Scylla with `replace-address-first-boot` by sidecar Container, the autoreplace InitContainer creates a file in a volume shared between the InitContainer and the sidecar Container. The file is then read by the sidecar Container.
+- To check whether its IP address already exists in gossip, the module uses ScyllaClient to request cluster's `status`.
+
+### Test Plan
+
+* e2e/integration tests
+     1.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for the `ScyllaCluster` to deploy
+          - Remove first node's PVC
+          - Remove first node's Pod
+          - Await PVC's and Pod's deletion
+          - Wait for the `ScyllaCluster` to redeploy
+          - Verify `ScyllaCluster`
+     2.
+          - Create a `ScyllaCluster` with 2 nodes
+          - Wait for `ScyllaCluster` to deploy
+          - Turn on maintenance mode, Exec `systemctl stop scylla && rm -rf /var/lib/scylla` in first node's Pod
+          - Remove first node's Pod
+          - Await Pod's deletion
+          - Wait for the `ScyllaCluster` to redeploy
+          - Verify `ScyllaCluster`
+
+### Upgrade / Downgrade Strategy
+
+This enhancement adds an InitContainer and a shared Volume to the Scylla Operator.
+It does not cause any drawbacks on a downgrade.
+
+### Version Skew Strategy
+
+This enhancement only requires changes in the Scylla Operator image. There is no version skew involved.
+
+## Implementation History
+
+- 1.5, enhancement proposal created
+
+## Drawbacks
+
+Introducing an additional InitContainer, as well as a new shared Volume is required.
+
+## Alternatives
```

Comment:
one alternative could be a detection from the local filesystem - either reading scylla files or creating a marker file when the node appears in bootstrap or becomes ready for the first time. also possibly talking to the local REST API.